### DISCUSSION TITLE
Which pem certificate should be used to enroll an Admin user

### DISCUSSION THREAD

**Discussion by deionut** [hyperledger/fabric-gateway#129] [Q&A]
Hi,

I am starting a new discussion related to [https://github.com/hyperledger/fabric-gateway/discussions/127](url)

I manage to successfully enroll the Admin user with this Java code:

```
    String pemStr = "copy-pasted from MyOrg_Admin file in the Node.js wallet directory";
    Properties caProperties = new Properties();
    caProperties.put("pemBytes", pemStr.getBytes());
		
    try
    {
        // caName and caUrl copy-pasted from connection.json["certificateAuthorities"][adr]["caName"] and ["url"]
        HFCAClient ca = HFCAClient.createNewInstance(caName, caUrl, caProperties);
        CryptoSuite cryptoSuite = CryptoSuite.Factory.getCryptoSuite();
        ca.setCryptoSuite(cryptoSuite);
        Enrollment adminEnrollment = ca.enroll("MyOrg_Admin", "password");
    }
    catch (Exception e) {
	    e.printStackTrace();
    }

```

The problem here is that the pem certificate I used is copy-pasted from the wallet directory imported with a separate Node.js client app. I clearly can not use this method of getting the pem certificate in a stand-alone Java client, so my question is where to get the PEM certificate from? I tried using the certificate found in the connection.json file under connection.json["certificateAuthorities"][adr]["tlsCACerts"]["pem"] but I get an exception thrown ( ERROR org.hyperledger.fabric_ca.sdk.HFCAClient - org.hyperledger.fabric.sdk.exception.CryptoException: Unable to add CA certificate to trust store. Error: java.io.IOException: Incomplete data ).

Another place I can think of to get the PEM is from the MyOrg_Admin_identity.json file that I downloaded from the IBM cloud blockchain web console when I created the user MyOrg_Admin; this file specifies a certificate but the string there is not in the PEM format (it doesn't start with -----BEGIN CERTIFICATE----- ); here is the format of MyOrg_Admin_identity.json:

```
{
    "name": "MyOrg_Admin",
    "type": "identity",
    "private_key": "blahblahblah==",
    "cert": "even longer blahblahblah that doesn't start with -----BEGIN CERTIFICATE-----=="
}

```
The Node.js client code I used to import the wallet didn't need any PEM certificate, it just needed the username and password of the Admin user and the ca name and url. If the Java code needs this certificate where do I get it from?

Thanks & regards,
Ionut

Later edit: my fault: the enrollment does NOT work with the above java code, I forgot to un-comment the ca.enroll() line, but the question still remains how to enroll an admin user and which PEM certificate to use ...

**Comment by bestbeforetoday** [comment]
This discussion area is for the new Fabric Gateway client API currently under development, not for the existing client SDKs.
